geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -53,10 +70,68 @@ class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
         NotFound(reason = s"Unknown ID $id")
       } else {
         contentType = "application/octet-stream"
-        response.setHeader("Content-Disposition", "attachment; filename=" + filename)
+        response.setHeader("Content-Disposition", "attachment;filename=" + filename)
 
         Ok(returnBytes)
       }
     }
   }
+
+  // TODO: Revisit configuration and persistence of configuration.
+  // https://geomesa.atlassian.net/browse/GEOMESA-958
+  // https://geomesa.atlassian.net/browse/GEOMESA-984
+  post("/ds/?") {
+    logger.debug("In ds registration method")
+
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+
+    if (ds == null) {
+      NotFound(reason = "Could not load data store using the provided parameters.")
+    } else {
+      this.synchronized {
+        abs = new AccumuloBlobStore(ds)
```

Comment:
I could just make it an atomic reference: https://twitter.github.io/scala_school/concurrency.html